src/st/cen/D600.c

```diff
@@ -613,7 +614,164 @@ s16 func_801904B8(Primitive* prim, s16 arg1) {
 }
 
 // Elevator when not moving (ID 1A)
-INCLUDE_ASM("asm/us/st/cen/nonmatchings/D600", EntityElevatorStationary);
+void EntityElevatorStationary(Entity* self) {
+    Entity* player = &PLAYER;
+    Primitive* prim;
+    s16 primIndex;
+    s16 posX, posY;
+    s16 temp;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_8018047C);
+        self->animCurFrame = 3;
+        self->zPriority = player->zPriority + 2;
+        CreateEntityFromCurrentEntity(E_ID1B, &self[-1]);
+        self[-1].params = 1;
+        CreateEntityFromCurrentEntity(E_ID1B, &self[-2]);
+        self[-2].params = 2;
+        primIndex = g_api.AllocPrimitives(PRIM_GT4, 12);
+        if (primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        prim = &g_PrimBuf[primIndex];
+        self->primIndex = primIndex;
+        self->ext.prim = prim;
+        self->flags |= FLAG_HAS_PRIMS;
+        prim->tpage = 0x12;
+        prim->clut = 0x223;
+        temp = prim->v1 = prim->u0 = prim->u2 = 0x28;
+        prim->v2 = prim->v3 = prim->u1 = prim->u3 = 0x38;
+        prim->v0 = temp;
+        prim->priority = 0x6B;
+        prim->blendMode = 8;
+        prim = prim->next;
+
+        while (prim != NULL) {
+            prim->tpage = 0x12;
+            prim->clut = 0x223;
+            prim->priority = 0x6A;
+            prim->blendMode = 8;
+            prim = prim->next;
+        }
+
+        if ((s16)(player->posY.i.hi + g_Camera.posY.i.hi) < 80) {
+            self->posY.i.hi = player->posY.i.hi;
+            player->posX.i.hi = self->posX.i.hi;
+            self->animCurFrame = 10;
+            g_Entities[1].ext.stub[0x00] = 1;
+            SetStep(3);
+        }
+        break;
+
+    case 1:
+        if (*(u8*)&self[-1].ext.stub[0x4]) {
```

Comment:
I think we'd want to update this when Entity1B is decompiled but I think it's fine for now